Error when running credit mining tests on win32

Got this one when running the tests:

testing random policy ... FAIL

```
======================================================================
FAIL: testing random policy
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win32\tribler\Tribler\Test\test_as_server.py", line 64, in check
    result = fun(*argv, **kwargs)
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win32\tribler\Tribler\Test\Core\CreditMining\test_creditmining.py", line 58, in test_random_policy
    self.assertEqual(3, len(ids_start), "Start failed %s vs %s" % (ids_start, torrents_start))
AssertionError: Start failed [4, 6] vs [{'num_seeders': 4, 'creation_date': 4, 'num_leechers': 3, 'metainfo': <Tribler.Test.Core.CreditMining.mock_creditmining.MockMeta object at 0x0636C190>}, {'num_seeders': 6, 'creation_date': 6, 'num_leechers': 5, 'metainfo': <Tribler.Test.Core.CreditMining.mock_creditmining.MockMeta object at 0x0636C810>}]
-------------------- >> begin captured logging << --------------------
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 52592
DEBUG   1477571218   SessionConfig:Session:125  Get random port 52592 for [general~minport]
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 19327
DEBUG   1477571218   SessionConfig:Session:125  Get random port 19327 for [dispersy~dispersy_port]
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 49277
DEBUG   1477571218   SessionConfig:Session:125  Get random port 49277 for [mainline_dht~mainline_dht_port]
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 35179
DEBUG   1477571218   SessionConfig:Session:125  Get random port 35179 for [video~port]
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 5772
DEBUG   1477571218   SessionConfig:Session:125  Get random port 5772 for [libtorrent~anon_listen_port]
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 44584
DEBUG   1477571218   SessionConfig:Session:125  Get random port 44584 for [tunnel_community~socks5_listen_ports~0]
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 26935
DEBUG   1477571218   SessionConfig:Session:125  Get random port 26935 for [tunnel_community~socks5_listen_ports~1]
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 50367
DEBUG   1477571218   SessionConfig:Session:125  Get random port 50367 for [tunnel_community~socks5_listen_ports~2]
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 41749
DEBUG   1477571218   SessionConfig:Session:125  Get random port 41749 for [tunnel_community~socks5_listen_ports~3]
DEBUG   1477571218   network_utils:Tribler.Core.Utilities.network_utils:34   Got a working random port 5062
DEBUG   1477571218   SessionConfig:Session:125  Get random port 5062 for [tunnel_community~socks5_listen_ports~4]
INFO    1477571218   sqlitecachedb:SQLiteCacheDB:135  begin page_size upgrade...
INFO    1477571218   sqlitecachedb:SQLiteCacheDB:139  ...end page_size upgrade
INFO    1477571218   sqlitecachedb:SQLiteCacheDB:159  Initializing new database...
INFO    1477571218   sqlitecachedb:SQLiteCacheDB:180  Reading database version...
INFO    1477571218   sqlitecachedb:SQLiteCacheDB:184  Current database version is 28
INFO    1477571218   sqlitecachedb:SQLiteCacheDB:202  Beginning the first transaction...
DEBUG   1477571218     mainlineDHT:Tribler.Core.DecentralizedTracking.mainlineDHT:29   DHT initialization True
DEBUG   1477571218     mainlineDHT:Tribler.Core.DecentralizedTracking.mainlineDHT:40   DHT running
--------------------- >> end captured logging << ---------------------
```

See https://jenkins.tribler.org/job/GH_Tribler_PR_tests_win32/3217/artifact/output/test_runners_output/0.err/*view*/

--- a/Tribler/Core/APIImplementation/LaunchManyCore.py
+++ b/Tribler/Core/APIImplementation/LaunchManyCore.py
@@ -186,7 +186,7 @@ class TriblerLaunchMany(TaskManager):
         return self.startup_deferred
 
     def on_tribler_started(self, subject, changetype, objectID, *args):
-        self.startup_deferred.callback(None)
+        reactor.callFromThread(self.startup_deferred.callback, None)
 
     def init(self):
         if self.dispersy:

--- a/Tribler/Test/Core/CreditMining/test_creditmining.py
+++ b/Tribler/Test/Core/CreditMining/test_creditmining.py
@@ -50,14 +50,16 @@ class TestBoostingManagerPolicies(TestAsServer):
         """
         testing random policy
         """
-        random.seed(0)
+        rdrwh = random.WichmannHill(0)
         policy = RandomPolicy(self.session)
+        policy.key = lambda _: rdrwh.random()
+
         torrents_start, torrents_stop = policy.apply(self.torrents, 6, force=True)
         ids_start = [torrent["metainfo"].get_infohash() for torrent in
                      torrents_start]
-        self.assertEqual(3, len(ids_start), "Start failed %s vs %s" % (ids_start, torrents_start))
+        self.assertEqual(1, len(ids_start), "Start failed %s vs %s" % (ids_start, torrents_start))
         ids_stop = [torrent["metainfo"].get_infohash() for torrent in torrents_stop]
-        self.assertEqual(2, len(ids_stop), "Stop failed %s vs %s" % (ids_stop, torrents_stop))
+        self.assertEqual(0, len(ids_stop), "Stop failed %s vs %s" % (ids_stop, torrents_stop))
 
     def test_seederratio_policy(self):
         """

